inspirehep/modules/records/api.py

```
@@ -24,45 +24,92 @@
+
+            files_src_record(InspireRecord): if passed, it will try to get the
+                files for the documents and figures from this record's files
+                iterator before downloading them, for example to merge existing
+                records.
+        """
+        files_src_record = kwargs.pop('files_src_record', None)
+        new_record = super(InspireRecord, cls).create(*args, **kwargs)
+        new_record.download_documents_and_figures(src_record=files_src_record)
+        new_record.commit()
+        return new_record
+
+    def update(self, *args, **kwargs):
+        """
+        Args:
+
+            files_src_record(InspireRecord): if passed, it will try to get the
+                files for the documents and figures from this record's files
+                iterator before downloading them, for example to merge existing
+                records.
+        """
+        files_src_record = kwargs.pop('files_src_record', None)
+        super(InspireRecord, self).update(*args, **kwargs)
+        self.download_documents_and_figures(
+            only_new=True,
+            src_record=files_src_record,
+        )
+
     def merge(self, other):
         """Redirect pidstore of current record to the other InspireRecord.
 
-        :param other: The record that self(record) is going to be redirected.
-        :type other: InspireRecord
+        Args:
+
+            other(InspireRecord): The record that self(record) is going to be
+                redirected.
         """
-        pids_deleted = PersistentIdentifier.query.filter(PersistentIdentifier.object_uuid == self.id).all()
-        pid_merged = PersistentIdentifier.query.filter(PersistentIdentifier.object_uuid == other.id).one()
+        pids_deleted = PersistentIdentifier.query.filter(
+            PersistentIdentifier.object_uuid == self.id
+        ).all()
+        pid_merged = PersistentIdentifier.query.filter(
+            PersistentIdentifier.object_uuid == other.id
+        ).one()
```

Nit: why these changes?

We don't enforce line length limits in INSPIRE (`E501` is ignored in https://github.com/inspirehep/inspire-next/blob/5f1a16289fde4753099ebd2febbddac5e9038c4a/pytest.ini#L25).

---

But my editor does, and complains, and pollutes the errors list. This way it's cleaner and I don't mistakenly ignore any errors at editing time.

---

Then your editor is misconfigured...

---

to some extent yes, but it seems to be more work to properly make it fetch the config from the current project than doing the small innocuous changes you see above.